### DISCUSSION TITLE
Recompile IDF components

### DISCUSSION THREAD

**Discussion by ghost** [micropython/micropython#10228] [ESP32] [answered]
Hi there,
I've modified an IDF component for esp32 (esp_eth) but it seems that Micropython compiles the old version of the file.

I'm compiling using `make BOARD=<<board>> all`

Is micropython cross-compiler using some already-compiled static library? How can I recompile that library in order to get those modifications? Thank you very much

**Comment by dhalbert** [comment, marked as answer]
Did you try `make clean...`? The make dependency checking may not extend into the esp-idf directory.

**Reply by ghost** [reply]
Yes, absolutely. I have tried with clean

I used a trick on Pycom boards. I compiled a project in order to get new static libraries and then copied them into my micropython project.

But on ESP SDK i cannot figure out where static libraries are coming from.

**Reply by jimmo** [reply]
> I used a trick on Pycom boards. I compiled a project in order to get new static libraries and then copied them into my micropython project.

What do you mean by this? There should be no need to do anything with static libraries.

You should just be able to modify IDF code and the build will work with the updated one. I do this pretty often.

**Reply by ghost** [reply]
Hi jimmo, thank you for your support.
I'm gonna explain what I'm doing at the moment in order to understand what's wrong with it.

I've modified my `micropython/ports/esp32/network_lan.c` and I've the following strings:

```
ESP_LOGE("network_lan", "DEFAULT_CONFIG");
esp_err_t esp_err = esp_eth_driver_install(&config, &self->eth_handle);
ESP_LOGE("network_lan", "esp_err");
```

I'm able to see the "DEFAULT_CONFIG" log message but I'm not able to see the "esp_err" log message.
So it might be a problem into `esp_eth_driver_install` function.

I've added a simple
`ESP_LOGE(TAG, "esp_eth_driver_install");`
at the beginning of the 'esp-idf/components/esp_eth/src/esp_eth.c' but I'm not able to see that message.
It seems that no modifications appen on the esp_eth IDF component, but I'M ABLE to see modifications on network_lan.c library

This is not the first time I encounter this problem.
Thank you

**Reply by git-n-pissed** [reply]
I've also been having problems with MP not detecting changes for ESP32 builds.

https://github.com/orgs/micropython/discussions/10246

**Comment by ghost** [comment]
Solved. Check it out for the correct ESP-IDF path and use `make clean` command to get the libraries updated.

Discussion can be closed
